packages/lib-classifier/src/plugins/tasks/models/Annotation.js

```diff
@@ -1,13 +1,30 @@
-import { types } from 'mobx-state-tree'
+import cuid from 'cuid'
+import { getSnapshot, types } from 'mobx-state-tree'
 
 const Annotation = types.model('Annotation', {
   id: types.identifier,
   task: types.string,
   taskType: types.string
 })
+  .preProcessSnapshot(snapshot => {
+    const newSnapshot = Object.assign({}, snapshot)
```

Comment:
Just checking, is this necessary because the serialization is immutable? Maybe to make this clear, the variable should be named `mutableSnapshot`?

Reply:
Yeah, you can't change `snapshot`. You have to create a completely new object.

>Note 2: The preProcessSnapshot transformation must be pure; it should not modify its original input argument!**_

https://mobx-state-tree.js.org/overview/hooks